packages/block-editor/src/components/block-lock/use-block-lock.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+/**
+ * Return details about the block lock status.
+ *
+ * @param {string}  clientId  The block client Id.
+ * @param {boolean} checkRoot Optional. Use root client ID when checking lock status.
+ *
+ * @return {Object} Block lock status
+ */
+export default function useBlockLock( clientId, checkRoot = false ) {
+	return useSelect(
+		( select ) => {
+			const {
+				canMoveBlock,
+				canRemoveBlock,
+				canLockBlockType,
+				getBlockName,
+				getBlockRootClientId,
+			} = select( blockEditorStore );
+			const rootClientId = checkRoot
+				? getBlockRootClientId( clientId )
+				: null;
+
+			return {
+				canMove: canMoveBlock( clientId, rootClientId ),
+				canRemove: canRemoveBlock( clientId, rootClientId ),
+				canLockBlock: canLockBlockType( getBlockName( clientId ) ),
```

Comment:
Minor thing, but I think it'd be good to name it `canLock` to match the naming of the other properties.

Reply:
I was thinking about this change as well. Thanks for the confirmation.

Reply:
Updated in [b47d36b](https://github.com/WordPress/gutenberg/pull/40210/commits/b47d36bf82ddbe66f0b23d5bdcb8613ae495e68a).